src/qml/editorwidgets/relationeditors/relation_editor.qml

```diff
@@ -13,134 +13,147 @@ EditorWidgetBase {
     id: relationEditor
 
     property int itemHeight: 40
+    property int maximumVisibleItems: 4
+
+    Component.onCompleted: {
+      if ( currentLayer && currentLayer.customProperty('QFieldSync/relationship_maximum_visible') !== undefined ) {
+        maximumVisibleItems = JSON.parse(currentLayer.customProperty('QFieldSync/relationship_maximum_visible'))[relationId]
+      } else {
+        maximumVisibleItems = 4
+      }
+    }
 
     // because no additional addEntry item on readOnly (isEnabled false)
-    height: isEnabled
+    height: (isEnabled
             ? referencingFeatureListView.height + itemHeight
-            : Math.max( referencingFeatureListView.height, itemHeight)
+            : Math.max( referencingFeatureListView.height, itemHeight)) + 10
```

Comment:
We should do something, this magic numbers, `10` in this case scare me a lot, I guess this is some margin now?